workitem/workitem_repository.go

```diff
@@ -17,10 +17,15 @@ import (
 	uuid "github.com/satori/go.uuid"
 )
 
+const orderValue = 1000
+const above = "above"
+const below = "below"
+
 // WorkItemRepository encapsulates storage & retrieval of work items
 type WorkItemRepository interface {
 	Load(ctx context.Context, ID string) (*app.WorkItem, error)
 	Save(ctx context.Context, wi app.WorkItem) (*app.WorkItem, error)
+	Reorder(ctx context.Context, position *app.WorkItemReorderPosition, wi app.WorkItem) (*app.WorkItem, error)
```

Comment:
Because you're passing REST objects down into the DB layer. You are now tightly coupling the layout of the REST API with the domain/db layer. We had the same with app.WorkItem vs app.WorkItem2. And we have the same with WorkItemLinks etc etc. It creates some couplings that cause some very odd behavior. Like we can't store a WorkItemLink without having a HTTP Request Context.

The Reorder function could easily be:

    Reorder(ctx contextContext, direction Direction, target uuid.UUID, wi app.WorkItem)